subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileToolchainIntegrationTest.groovy

```diff
@@ -475,12 +676,49 @@ public class Foo {
         JavaVersion.VERSION_17  | "[deprecation] foo() in Foo has been deprecated"
```

Comment:
These Java versions also don't seem to be future-proof. It seems like we want to test on JDK 8 and any higher installed JDK. Can we encode this in the where clause somewhere? Or in the test itself?